yt_dlp/extractor/youtube.py

```diff
@@ -3388,21 +3417,39 @@ def extract_header(contents):
                 break
             return _continuation
 
-        def extract_thread(contents):
+        def extract_thread(contents, entity_payloads):
             if not parent:
                 tracker['current_page_thread'] = 0
             for content in contents:
                 if not parent and tracker['total_parent_comments'] >= max_parents:
                     yield
                 comment_thread_renderer = try_get(content, lambda x: x['commentThreadRenderer'])
-                comment_renderer = get_first(
-                    (comment_thread_renderer, content), [['commentRenderer', ('comment', 'commentRenderer')]],
-                    expected_type=dict, default={})
 
-                comment = self._extract_comment(comment_renderer, parent)
+                # old comment format
+                if not entity_payloads:
+                    comment_renderer = get_first(
+                        (comment_thread_renderer, content), [['commentRenderer', ('comment', 'commentRenderer')]],
+                        expected_type=dict, default={})
+
+                    comment = self._extract_comment_old(comment_renderer, parent)
+
+                # new comment format
+                else:
+                    view_model = (
+                        traverse_obj(comment_thread_renderer, ('commentViewModel', 'commentViewModel', {dict}))
+                        or traverse_obj(content, ('commentViewModel', {dict})))
+                    if not view_model:
+                        continue
+                    comment_keys = list(filter(None, [view_model.get('commentKey'), view_model.get('toolbarStateKey')]))
+                    entities = traverse_obj(entity_payloads, lambda _, v: v['entityKey'] in comment_keys)
```

Comment:
`commentKey` and `toolbarStateKey` appear to always be the same value (and logically should be); the latter is just a fallback, right?

in that case, this would be simpler:

```suggestion
                    view_model = (
                        traverse_obj(comment_thread_renderer, ('commentViewModel', 'commentViewModel', {dict}))
                        or traverse_obj(content, ('commentViewModel', {dict})))
                    comment_key = traverse_obj(view_model, 'commentKey', 'toolbarStateKey', expected_type=str)
                    if not comment_key:
                        continue
                    entities = traverse_obj(entity_payloads, lambda _, v: v['entityKey'] == comment_key)
```

Reply:
@bashonly 

> commentKey and toolbarStateKey appear to always be the same value (and logically should be); the latter is just a fallback, right?

Wow... hold on just a second. No they're not. They're different values, and they're BOTH necessary. One key is to obtain the comment entity, another to obtain the entity containing engagementToolbarStateEntityPayload.

Here's an example. The difference - one letter - is highlighted in bold:
commentKey: `EhpVZ3dJYjFXSkZSbUN3VW9td2I5NEFhQUJBZyA`**o**`KAE%3D`
toolbarStateKey: `EhpVZ3dJYjFXSkZSbUN3VW9td2I5NEFhQUJBZyA`**s**`KAE%3D`

Reply:
ok yeah this is why I asked